torch/_higher_order_ops/while_loop.py

```diff
@@ -21,31 +23,47 @@
 
 
 class WhileLoopOp(HigherOrderOperator):
-    def __call__(self, cond_fn, body_fn, operands):
+    def __call__(self, cond_fn, body_fn, carried_inputs, additional_inputs):
```

Comment:
What are the chances you can make this positional-only lol.

e.g.:
```
def __call__(self, cond_fn, body_fn, carried_inputs, additional_inputs, /)
```

Reply:
Yeah, I think we can do that.

I just found previously, we're not using this WhileLoopOp but the plain HigherOrderOp. Updated it to use the WhileLoopOp for checking the inputs. Will add more tests for invalide inputs as a follow-up.